spec/cc/engine/violation_spec.rb

```diff
@@ -0,0 +1,48 @@
+require "spec_helper"
+require "cc/engine/duplication"
+
+module CC::Engine::Analyzers
+  RSpec.describe Violation do
+    let(:source) do
+      <<-SOURCE
+        begin
+          foo
+        rescue SyntaxError => e
+          Jekyll.logger.warn "YAML Exception reading \#{File.join(base, name)}: \#{e.message}"
+        rescue Exception => e
+          Jekyll.logger.warn "Error reading file \#{File.join(base, name)}: \#{e.message}"
+        end
+      SOURCE
+    end
+    let(:flay) do
+      Flay.new({
+        diff: false,
+        mass: CC::Engine::Analyzers::Ruby::Main::DEFAULT_MASS_THRESHOLD,
+        summary: false,
+        verbose: false,
+        number: true,
+        timeout: 10,
+        liberal: false,
+        fuzzy: false,
+        only: nil,
+      })
+    end
+
+    describe "violation location" do
+      it "gets appropriate locations for rescue blocks" do
+        sexp = RubyParser.new.process(source, "file.rb")
+        flay.process_sexp(sexp)
+
+        report = flay.analyze[0]
+        violation = Violation.new(10, report, flay.hashes[report.structural_hash])
+
+        location = violation.send(:format_location)
+        other_location = violation.send(:format_other_locations)[0]
+
+        expect(location).not_to eq other_location
+        expect(location[:lines]).to eq({:begin => 3, :end => 7})
```

Comment:
The overlap between these two nodes is a little weird, but makes sense from a parser sense: syntactically, a `rescue` expr doesn't end until the `end`: the next `rescue` is a distinct node, but the two do overlap in terms of lines.

I've opted not to do anything about that right now because 1) it's not a super common case for code duplication and 2) I don't have a good answer for a generalizable automatic remediation. We could just drop the "overlapping" lines from the bigger node (i.e. the above location would become `{:begin => 3, :end => 4}`, but I'm not confident there aren't other cases of node overlap that would result in even more nonsensical display.